src/Project/Project.router.ts

```diff
@@ -84,6 +84,11 @@ export class ProjectRouter extends Router {
       server.handleRequest(this.deleteProject)
     )
 
+    this.router.get(
+      '/projects/:id/public',
```

Comment:
I think we can maybe keep `withProjectExists` here, to be at least more consistent with the other endpoints.

If not, maybe it's a good idea to check the existence of `Project.findOne({ id, is_public: true })` (below) before returning it, so we can return a `404` like the other endpoints.